Scripts/Editor/Configuration/WitConfigurationEditor.cs

```diff
@@ -20,65 +20,111 @@ public class WitConfigurationEditor : Editor
 
     private int selectedToolPanel;
 
-    private readonly string[] toolPanelNames = new []
+    private readonly string[] toolPanelNames = new[]
     {
+        "Application",
         "Intents",
         "Entities"
     };
 
-    private const int TOOL_PANEL_INTENTS = 0;
-    private const int TOOL_PANEL_ENTITIES = 1;
+    private readonly string[] toolPanelNamesWithoutAppInfo = new[]
+    {
+        "Intents",
+        "Entities"
+    };
+
+    private const int TOOL_PANEL_APP = 0;
+    private const int TOOL_PANEL_INTENTS = 1;
+    private const int TOOL_PANEL_ENTITIES = 2;
 
     private Editor applicationEditor;
     private Vector2 scroll;
+    private bool appConfigurationFoldout;
 
     private bool IsTokenValid => !string.IsNullOrEmpty(configuration.clientAccessToken) &&
                                  configuration.clientAccessToken.Length == 32;
 
-    private void OnEnable()
+    public void OnEnable()
     {
         configuration = target as WitConfiguration;
-        configuration.Update();
+        configuration.UpdateData();
     }
 
     public override void OnInspectorGUI()
```

Comment:
OnInspectorGUI is a bit difficult for me to follow, there's a lot of auth code in there, although I guess that it's auth-related GUI. Not sure on patterns for compartmentalizing GUI code in Unity, if there are any.